index.js

```diff
@@ -29,6 +31,9 @@ const fastifyAutoload = async function autoload (fastify, options) {
   const pluginsMeta = {}
   const hooksMeta = {}
 
+  const fastifyRegister = fastify.register
+  fastify.register = (plugin, options) => opts.encapsulate ? fastifyRegister(plugin, options) : fastifyRegister(fp(plugin), options)
```

Comment:
What is the `plugin` function is already a plugin?
We should check the `fn[Symbol.for('skip-override')]` value

Reply:
thanks @Eomm, addressed as part of @mcollina comment